techniques/system/common/1.0/update.st

```diff
@@ -321,6 +328,30 @@ bundle agent update_action
       action => immediate;
 }
 
+# Only persist classes once as it is a very expensive operation now.
+# Done outside of update_action as some actions are done in files promises during last pass.
+bundle agent rudder_persist_update_classes {
+  vars:
+    pass1.!pass2::
+       "clean_classes" string => "",
+                      classes => u_cancel_classes("${update_action.classes_to_persist}");
```

Comment:
this always clean the classes, even when no update has been made. It should be done only if it did the update

Reply:
oh indeed.....